lightning/src/ln/functional_test_utils.rs

```diff
@@ -853,13 +893,13 @@ macro_rules! expect_payment_failed {
 		assert_eq!(events.len(), 1);
 		match events[0] {
 			Event::PaymentFailed { ref payment_hash, rejected_by_dest, ref error_code, ref error_data } => {
-				assert_eq!(*payment_hash, $expected_payment_hash);
-				assert_eq!(rejected_by_dest, $rejected_by_dest);
-				assert!(error_code.is_some());
-				assert!(error_data.is_some());
+				assert_eq!($expected_payment_hash, *payment_hash, "unexpected payment_hash");
+				assert_eq!($rejected_by_dest, rejected_by_dest, "unexpected rejected_by_dest value");
+				assert!(error_code.is_some(), "expected error_code.is_some() = true");
+				assert!(error_data.is_some(), "expected error_data.is_some() = true");
 				$(
-					assert_eq!(error_code.unwrap(), $expected_error_code);
-					assert_eq!(&error_data.as_ref().unwrap()[..], $expected_error_data);
+					assert_eq!($expected_error_code, error_code.unwrap(), "unexpected error code");
+					assert_eq!($expected_error_data, &error_data.as_ref().unwrap()[..], "unexpected error data");
 				)*
```

Comment:
Any reason to change the ordering? I tend to think leaving the expected last is more readable in code. Not sure if the error message is suppose to correspond with either side despite being printed on the same line as the second.

Reply:
Interesting, I find `expected: {}, actual: {}` more readable (I think I made this change originally).

Reply:
FWIW, my reading is "assert that X has expected value Y".